i3pystatus/core/io.py

```diff
@@ -75,6 +93,62 @@ def read_line(self):
 
         return self.proto[min(self.n, len(self.proto) - 1)]
 
+    @classmethod
+    def compute_treshold_interval(cls):
+        """
+        Current method is to compute average from all intervals.
+
+        Make sure to call after all modules are registered.
+        """
+
+        intervals = [m.interval for m in cls.modules if hasattr(m, "interval")]
+        cls.treshold_interval = round(sum(intervals) / len(intervals))
```

Comment:
I think calls to this are "rare enough" that we don't need to additionally cache it with a class-level variable